### DISCUSSION TITLE
Controlled components do not surface Ref in errors even when ref is set

### DISCUSSION THREAD

**Discussion by sgarfinkel** [react-hook-form/react-hook-form#10162] [Q&A] [answered]
Looks like this may be intended, but it’s a big problem for me. The issue is here https://github.com/react-hook-form/react-hook-form/blob/master/src/useController.ts#L138. I need to get the actual HTMLElement because I want to scroll to the first error on the page, but I have no way of calculating which element is first because the error doesn’t actually hold the the Ref. Is this intended? Is there any workaround?

**Comment by leapful** [comment, marked as answer]
@sgarfinkel 

react-hook-form already supports option `shouldFocusError` by default to focus and scroll automatically to first field error.

If you are using react-hook-form on web and it does not work, please provide a codesandbox url to re-produce the issue.

If you are using something like React Native which might not support scroll into view on focus element, you have to implement the behavior manually. The object `formState.errors` is already a key error object where key is field name and you, the developer, already knows what is the order of you input fields on UI, so you just need to check if list ordered of your input fields to see if it exists in `formState.errors` object then perform scroll into it.

Something like:

const LIST_ORDER_UI = ['A', 'B', 'C'];

```
const { formState: { errors } } = useForm();

useEffect(() => {
   for (const name of LIST_ORDER_UI) {
      if (errors[name]) {
         document.querySelector(`[name=${name}]`)?.scrollIntoView();
      }
   }
}, [errors]);

return <form>
<input name="A" >
<input name="B" >
<input name="C" >
</form>
```
